File: dev-docs/snippets/blob-buckets/delete-bucket.swift
import JsBaoClient

// Delete a bucket and every blob inside it. Swift returns a typed
// `BlobDeletedResult` mirroring the JS `{ deleted: boolean }` shape.
func deleteBucket(client: JsBaoClient, bucketIdOrKey: String) async throws {
  // #region example
  let result = try await client.blobBuckets.deleteBucket(bucketIdOrKey: bucketIdOrKey)
  let deleted = result.deleted
  // #endregion example
  _ = deleted
}

File: dev-docs/snippets/blob-buckets/delete-bucket.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Delete a bucket and every blob inside it. Resolves to `{ deleted: boolean }`.
export async function deleteBucket(client: JsBaoClient, bucketIdOrKey: string) {
  // #region example
  const { deleted } = await client.blobBuckets.deleteBucket(bucketIdOrKey);
  // #endregion example
  return deleted;
}
